src/test/java/org/opentripplanner/middleware/cdp/ConnectedDataPlatformTest.java

```diff
@@ -0,0 +1,241 @@
+        Set<LocalDate> betweenDays = DateTimeUtils.getDatesBetween(
+            getDatePlusNumberOfDays(sevenDaysAgo,1),
+            new Date()
+        );
+        TripHistoryUpload tripHistoryUpload = new TripHistoryUpload(sevenDaysAgo);
+        Persistence.tripHistoryUploads.create(tripHistoryUpload);
+        ConnectedDataManager.stageUploadDays();
+        assertEquals(
+            betweenDays.size(),
+            Persistence.tripHistoryUploads.getCountFiltered(Filters.gt("uploadDate", sevenDaysAgo))
+        );
+    }
+
+    /**
+     * Add an OTP user with 'storeTripHistory' set to true and related trip requests/summaries. Via the API update the
+     * 'storeTripHistory' to false and confirm that the trip history is removed and the appropriate days are flagged for
+     * updating.
+     */
+    @Test
+    public void canRemoveTripHistoryViaAPI() throws Exception {
+        assumeTrue(IS_END_TO_END);
+        ConnectedDataManager.IS_TEST = true;
+
+        // Set back stop. This allows dates after this to trigger an upload.
+        Date twentyDaysInThePast = getStartOfDay(getDateMinusNumberOfDays(new Date(), 20));
+        TripHistoryUpload tripHistoryUpload = new TripHistoryUpload(twentyDaysInThePast);
+        tripHistoryUpload.status = TripHistoryUploadStatus.COMPLETED.getValue();
+        Persistence.tripHistoryUploads.create(tripHistoryUpload);
+
+        // Create OTP user and trip data.
+        otpUser = PersistenceTestUtils.createUser("test@example.com");
+        Date tenDaysInThePast = getStartOfDay(getDateMinusNumberOfDays(new Date(), 10));
+        tripRequest = PersistenceTestUtils.createTripRequest(otpUser.id, tenDaysInThePast);
+        tripSummary = PersistenceTestUtils.createTripSummary(tripRequest.id, tenDaysInThePast);
+
+        // Update 'storeTripHistory' value.
+        otpUser.storeTripHistory = false;
+        mockAuthenticatedRequest(
+            String.format("%s/%s", OTP_USER_PATH, otpUser.id),
+            JsonUtils.toJson(otpUser),
+            otpUser,
+            HttpMethod.PUT
+        );
+        // Only expecting one trip history upload entry matching the date the trip request and summary were made.
+        assertEquals(
+            1,
+            Persistence.tripHistoryUploads.getCountFiltered(Filters.eq("uploadDate", tenDaysInThePast))
+        );
+
+        // Trip request and summary are removed as part of the 'removeUsersTripHistory' method. Setting these to null
+        // prevents a delete error under the tidy-up process.
+        tripRequest = null;
+        tripSummary = null;
+
+        // Set the zip file name and temp file name so they are both removed as part of the tidy-up process.
+        zipFileName = getFileName(tenDaysInThePast, ConnectedDataManager.ZIP_FILE_NAME_SUFFIX);
+        tempFile = String.join(
+            "/",
+            FileUtils.getTempDirectory().getAbsolutePath(),
+            zipFileName
+        );
```

Comment:
Test cases shouldn't have side effects on other test cases if at all possible. Is there a way these variables can be made to have a scope active only within this method?